### PR TITLE
Add comment

### PR DESCRIPTION
**Pull request by rosiefhiggins**

### CODE REVIEW

File: src/Components/CommentAdder.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { postComment } from "../api";
+import { useState } from "react";
+
+
+const CommentAdder = ({setComments, review_id}) => {
+    const [newComment, setNewComment] = useState("")
+    const [err, setError]=useState(null)
```
Comment (lewispricey) [inline src/Components/CommentAdder.jsx:7]: Really like the use of state here to hold an error

File: src/Components/CommentAdder.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    if (err) {
+        return <main><p>Error uploading comment!</p></main>
+    }
+    
+
+    return (
+        <form  onSubmit={handleSubmit}>
+        <textarea id="new-comment" value={newComment} onChange={(e)=> setNewComment(e.target.value)} />
```
Comment (lewispricey) [inline src/Components/CommentAdder.jsx]: Great choice with a text area here instead of an input 👍

File: src/Components/CommentAdder.jsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            "created_at": "Just now",
+        }
+        setComments((currComments)=>{
+            return [comment, ...currComments]
+        })
+        postComment(review_id, newComment).then(()=>{
+            setNewComment("")
+        })
+        .catch((err)=>{
+            setError(true)
```
Comment (lewispricey) [inline src/Components/CommentAdder.jsx:30]: If there is an error it would be good to remove the post you've optimisticaly rendered.

**Review by lewispricey — CHANGES_REQUESTED** [review]
Great work on this one so far Rosie!

Just a few things I noticed while testing:
- Please could you add in a way to un-optimisticaly render a new comment if the request fails. 
- I can currently post a comment with nothing in the text area, could you add in a way to stop this being posted.

Please raise another PR once you've made the changes above.

**Review by leekli — APPROVED** [review]
Hi Rosie,

So I can see you've implemented what Lewis asked.

My only point I would add... if there's an empty comment and the user tries to post it, you'd want to give them an indication they've done something wrong. Currently it just does nothing.

I would recommend that in the `<textarea>` tag you add an attribute called `required`, this will display an error to the user for you which will say something has to be in the form.

Here is the w3schools docs on how `required` works: [https://www.w3schools.com/tags/att_required.asp](url)

I'm to approve this ticket and for you to merge it, but I would strongly recommend you add the required tag, and test it works, before you merge it - no need to submit a new PR for it.

**Pull request merged**